core/fetcher.go

```diff
@@ -145,45 +211,133 @@ func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan types
 		defer close(f.doneCh)
 		defer close(signedBlockCh)
 		for {
+			resp, err := subscription.Recv()
+			if err != nil {
+				// case where the context was not canceled but still received an error
+				if ctx.Err() == nil {
+					log.Errorw("fetcher: error receiving new height", "err", err.Error())
+				}
+				return
+			}
+			signedBlock, err := f.GetSignedBlock(ctx, &resp.Height)
+			if err != nil {
+				log.Errorw("fetcher: error receiving signed block", "height", resp.Height, "err", err.Error())
+				return
+			}
```

Comment:
Lets add per-call timeout. Otherwise we might get stuck on one of the calls indefinitely

Also would need to handle canceled context here

Reply:
92c4f942f62503ef41cb40af50a835f463cea216

Reply:
92c4f942f62503ef41cb40af50a835f463cea216